utils/environment.go

```diff
@@ -68,13 +68,13 @@ type envEnvelope struct {
 	Languages  LanguageList `json:"languages"`
 }
 
-func ReadEnvironment(data *json.RawMessage) (*environment, error) {
+func ReadEnvironment(data json.RawMessage) (*environment, error) {
```

Comment:
I think this change is ok because it is just reading, but that is not actually true about slices in go. They are essentially a pointer to an array which has length information, but that pointer can change if you append to the slice, which is why the primitive `append(slice, newVal)` has to be assigned to `slice`.